mutiny-core/src/nodemanager.rs

```diff
@@ -1803,6 +1865,104 @@ impl<S: MutinyStorage> NodeManager<S> {
         Ok(mutiny_channels)
     }
 
+    fn get_scb_key(&self) -> SecretKey {
+        let seed = self.mnemonic.to_seed("");
+        let xprivkey = ExtendedPrivKey::new_master(self.network, &seed).unwrap();
+        let path = DerivationPath::from_str("m/444'/444'/444'").unwrap();
+        let context = Secp256k1::new();
+
+        xprivkey.derive_priv(&context, &path).unwrap().private_key
+    }
+
+    /// Creates a static channel backup for all the nodes in the node manager.
+    /// The backup is encrypted with the SCB key.
+    pub async fn create_static_channel_backup(&self) -> EncryptedSCB {
+        let nodes = self.nodes.lock().await;
+        let mut backups: HashMap<PublicKey, (NodeIndex, StaticChannelBackup)> = HashMap::new();
+        for (_, node) in nodes.iter() {
+            let scb = node.create_static_channel_backup();
+            backups.insert(node.pubkey, (node.node_index(), scb));
+        }
+
+        let peers = get_all_peers(&self.storage).unwrap_or_default();
+
+        let peer_connections = peers
+            .into_iter()
+            .filter_map(|(n, p)| p.connection_string.map(|str| (n.as_pubkey().unwrap(), str)))
+            .collect::<HashMap<_, _>>();
+
+        let scb = StaticChannelBackupStorage {
+            backups,
+            peer_connections,
+        };
+
+        // encrypt
+        let encryption_key = self.get_scb_key();
+        scb.encrypt(&encryption_key)
+    }
+
+    /// Takes an encrypted static channel backup and recovers the channels from it.
+    /// If the backup is encrypted with a different key than the current key, it will fail.
+    pub async fn recover_from_static_channel_backup(
+        &self,
+        scb: EncryptedSCB,
+    ) -> Result<(), MutinyError> {
+        // decrypt
+        let encryption_key = self.get_scb_key();
+        let scb = scb.decrypt(&encryption_key)?;
+
+        let max_index = scb
+            .backups
+            .values()
+            .map(|(node_index, _)| node_index.child_index)
+            .max()
+            .unwrap_or(0);
+
+        let current_index = self
+            .node_storage
+            .lock()
+            .await
+            .nodes
+            .values()
+            .map(|n| n.child_index)
+            .max()
+            .unwrap_or(0);
+
+        let needed_nodes = max_index - current_index;
+        if needed_nodes > 0 {
+            log_info!(
+                self.logger,
+                "Creating {needed_nodes} new nodes to recover from backup"
+            );
+            for _ in 0..needed_nodes {
+                self.new_node().await?;
```

Comment:
> Should this start up without making any connections to peers? I just added that ability.

It worked without it, could be worth adding.

> There's nothing stopping the recovery happening from an existing state, right?

Yeah

>  If there is a channel that is messed up, the node will panic if there's a bad state when it connects to a peer.

That is a good point. we probably want it to connect with a different state so that if we are in a bad state, it doesn't use it..